### PR TITLE
Displays proper error message for network errors

### PR DESCRIPTION
**Pull request by sam33rdhakal**
Updates
* Displays no internet connection for network error message
* Displays something went wrong when the error message is blank
* Updates exhibition search service to include new `image_url` attribute

### CODE REVIEW

File: db/src/main/kotlin/edu/artic/ResultExtensions.kt [hunk 1]
```diff
@@ -16,19 +16,21 @@ import org.json.JSONObject
...
     lateinit var errorMessage: String
     try {
         this.response().errorBody()?.also { errorBody ->
             val errorJSON = JSONObject(errorBody.string())
-            errorMessage = errorJSON.optString("error")
+            val message = errorJSON.optString("error")
                     .orIfNullOrBlank(errorJSON.optString("message"))
                     .orIfNullOrBlank(errorJSON.optString("Message"))
                     .orIfNullOrBlank(errorJSON.optString("detail"))
-                    ?: "Something went wrong"
```
Comment (Cliabhach) [inline db/src/main/kotlin/edu/artic/ResultExtensions.kt]: What are your thoughts on using `.orIfNullOrBlank(SOME_THING_WENT_WRONG)` here?

File: db/src/main/kotlin/edu/artic/ResultExtensions.kt [hunk 2]
```diff
@@ -16,19 +16,21 @@ import org.json.JSONObject
  * Looks for error, message and detail keys in order.
  */
 fun Result<*>.getErrorMessage(): String {
+    val SOME_THING_WENT_WRONG : String = "Something went wrong"
```
Comment (Cliabhach) [inline db/src/main/kotlin/edu/artic/ResultExtensions.kt]: `SOME_THING_WENT_WRONG` -> `SOMETHING_WENT_WRONG`

File: db/src/main/kotlin/edu/artic/ResultExtensions.kt (hunk 1 above)
Comment (sam33rdhakal) [reply db/src/main/kotlin/edu/artic/ResultExtensions.kt]: `orIfNullOrBlank` returns `String?` and `errorMessage` is of type `String`.

Comment (sam33rdhakal) [reply db/src/main/kotlin/edu/artic/ResultExtensions.kt]: I can use bang operator here if i use `.orIfNullOrBlank(SOME_THING_WENT_WRONG)`
```
val message = errorJSON.optString("error")
                    .orIfNullOrBlank(errorJSON.optString("message"))
                    .orIfNullOrBlank(errorJSON.optString("Message"))
                    .orIfNullOrBlank(errorJSON.optString("detail"))
                    .orIfNullOrBlank(SOMETHING_WENT_WRONG)
            errorMessage = message!!
```

File: db/src/main/kotlin/edu/artic/ResultExtensions.kt (hunk 2 above)
Comment (sam33rdhakal) [reply db/src/main/kotlin/edu/artic/ResultExtensions.kt]: Updated.

File: db/src/main/kotlin/edu/artic/ResultExtensions.kt (hunk 1 above)
Comment (sam33rdhakal) [reply db/src/main/kotlin/edu/artic/ResultExtensions.kt]: Ok.

**Pull request merged**